### PR TITLE
Added `get_keys_for_plot` daemon server API

### PR DESCRIPTION
**Pull request by ChiaMineJP**
This PR is a partial PR cut from cmj.plot2_0 branch.

### CODE REVIEW

File: chia/daemon/server.py [hunk 1]
```diff
@@ -567,28 +592,15 @@ async def get_routes(self, websocket: WebSocketResponse, request: Dict[str, Any]
         return response
 
     async def get_wallet_addresses(self, websocket: WebSocketResponse, request: Dict[str, Any]) -> Dict[str, Any]:
-        all_keys = Keychain().get_keys(include_secrets=True)
         fingerprints = request.get("fingerprints", None)
```
Comment (arvidn) [inline chia/daemon/server.py:595]: I suppose if the "fingerprints" value is not a `List[uin32]`, you get a poor error message. We assume that's the type in the call to `_get_keys_by_fingerprints()`

**Review by arvidn — COMMENTED** [review]
looks reasonable. do you think it's realistic to have tests? I don't think we have any infrastructure for it

File: chia/daemon/server.py (hunk 1 above)
Comment (ChiaMineJP) [reply chia/daemon/server.py:597]: Added assertion.

File: tests/core/daemon/test_daemon.py [hunk 2]
```diff
@@ -596,6 +609,74 @@ def get_keys_no_secrets(self, include_secrets):
...
+    ),
+)
+@pytest.mark.asyncio
+async def test_get_keys_for_plot_error(
+    mock_daemon_with_config_and_keys,
+    monkeypatch,
+    case: KeysForPlotCase,
+):
+    daemon = mock_daemon_with_config_and_keys
+    with pytest.raises(Exception):
```
Comment (arvidn) [inline tests/core/daemon/test_daemon.py:682]: what's the error we expect here? maybe you can add a `match="..."` to that `raises()` call

File: tests/core/daemon/test_daemon.py [hunk 3]
```diff
@@ -617,12 +617,16 @@ def get_keys_no_secrets(self, include_secrets):
...
+                    "pool_public_key": "b1b0d8e777e3b1e79f33bc8707c74c8e4907fd8c3da870d7"
+                    "81803f647b17f3cc6250f09362f25699eb0ee385c57a2e01",
                 },
                 test_key_data_2.fingerprint: {
-                    "farmer_public_key": "a96a194fe76043b4867fa154b1fcda6709257699a720cc10af68a8414d9bf08cee0945becf9b36a1c76d5b04d379bc7f",
-                    "pool_public_key": "837646f1378b1194aa1054b8869dcea5006cb173730ceeb90e9b9bb7f5d73e481775d20a1726fec3ef140c5b9cdbd4b3",
+                    "farmer_public_key": "a96a194fe76043b4867fa154b1fcda6709257699a720cc10"
+                    "af68a8414d9bf08cee0945becf9b36a1c76d5b04d379bc7f",
+                    "pool_public_key": "837646f1378b1194aa1054b8869dcea5006cb173730ceeb9"
+                    "0e9b9bb7f5d73e481775d20a1726fec3ef140c5b9cdbd4b3",
```
Comment (arvidn) [inline tests/core/daemon/test_daemon.py:629]: it's not easy to tell that these numbers make sense and that the test is correct. can you add some comments (or compute these in some way that makes it clear), to explain to the reader of what these test cases exercise any where these numbers come from?

Comment (ChiaMineJP) [reply tests/core/daemon/test_daemon.py:629]: I changed the code to derive keys from the original test private keys.

File: tests/core/daemon/test_daemon.py (hunk 2 above)
Comment (ChiaMineJP) [reply tests/core/daemon/test_daemon.py:682]: Done

**Comment by emlowe** [comment]
This seems fine, but I don't like the name that much `get_keys_for_plot` would suggest to me that I provide a PLOT as a parameter to the call and get back the keys used for the plot.

However, I believe this call is returning the farmer and pool public keys for the given fingerprint. So it returns the public keys one would use for plotting (it is the same farmer and pool pk as `chia keys show`)

Some alternative names:
get_plotting_public_keys
get_plotting_pks
get_farmer_pool_public_keys
get_farmer_pool_pks
get_pks_for_plotting
get_keys_for_plotting

File: tests/core/daemon/test_daemon.py (hunk 3 above)
Comment (emlowe) [reply tests/core/daemon/test_daemon.py]: Resoving comment per code change

**Comment by ChiaMineJP** [comment]
@emlowe I picked up `get_keys_for_plotting`.

**Review by emlowe — APPROVED** [review]

**Pull request merged**